### PR TITLE
Quickstarts: servlet-filterlistener and servlet-async

### PR DESCRIPTION
**Pull request by csadilek**
This pull request contains Jonathan Fuerth's and my quickstarts (servlet-filterlistener and servlet-async).

servlet-async shows how to detach the execution of a resource intensive task from the request processing thread, so the thread is free to serve other client requests. To properly demonstrate this feature, the servlet has to hand off a task to another thread (of a smaller pool) which will generate the response asynchronously. To keep things simple (less code) for the quickstart, we decided against managing our own thread pool/executors and instead also show the use of @Asynchronous EJB invocations, which provide exactly that.

servlet-filterlistener shows the use of a servlet filter to remove vowels from request parameter values and a servlet listener that logs all parameters of HTTP requests.

### CODE REVIEW

File: servlet-filterlistener/src/main/java/org/jboss/as/quickstarts/servlet/filterlistener/FilterExampleServlet.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * <p>
+ * A simple servlet that exists as a target for Servlet Filters and Servlet Listeners.
+ * </p>
+ * 
+ * <p>
+ * The servlet is registered and mapped to /HelloServlet using the {@linkplain WebServlet
+ * @HttpServlet}. The {@link HelloService} is injected by CDI.
```
Comment (pmuir) [inline servlet-filterlistener/src/main/java/org/jboss/as/quickstarts/servlet/filterlistener/FilterExampleServlet.java]: This comment seems out of date?

File: servlet-filterlistener/src/main/java/org/jboss/as/quickstarts/servlet/filterlistener/VowelRemoverFilter.java [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+      @Override
+      public String[] getParameterValues(String name) {
+        return filteredParams.get(name);
+      }
+    };
+
+    // Some notes on other filter use cases:
+    // 1. We could have also wrapped the response object if we wanted to capture
+    //    or manipulate the output
+    // 2. If we just wanted to examine the session
```
Comment (pmuir) [inline servlet-filterlistener/src/main/java/org/jboss/as/quickstarts/servlet/filterlistener/VowelRemoverFilter.java]: Comment seems unfinished

File: servlet-filterlistener/src/main/java/org/jboss/as/quickstarts/servlet/filterlistener/VowelRemoverFilter.java [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.annotation.WebFilter;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletRequestWrapper;
+import javax.servlet.http.HttpSession;
+
+/**
+ * A silly Servlet Filter that removes the letters a, e, i, o, and u (but not
+ * sometimes y) from all request parameter values. To achieve this, a wrapper is
```
Comment (pmuir) [inline servlet-filterlistener/src/main/java/org/jboss/as/quickstarts/servlet/filterlistener/VowelRemoverFilter.java]: Not sure about the comment about y?

**Comment by pmuir** [comment]
Couple of minor comments, otherwise excellent :-)

Can you do a rebase to remove the merge commits, and perhaps collapse this down into two commits, one for each example?

**Comment by csadilek** [comment]
OK thanks! We'll send an update soon.

File: servlet-filterlistener/src/main/java/org/jboss/as/quickstarts/servlet/filterlistener/VowelRemoverFilter.java (hunk 2 above)
Comment (jfuerth) [inline servlet-filterlistener/src/main/java/org/jboss/as/quickstarts/servlet/filterlistener/VowelRemoverFilter.java]: Hmm, perhaps overly brief compared to the others. I will include reasons for examining the session.

File: servlet-filterlistener/src/main/java/org/jboss/as/quickstarts/servlet/filterlistener/VowelRemoverFilter.java (hunk 3 above)
Comment (jfuerth) [inline servlet-filterlistener/src/main/java/org/jboss/as/quickstarts/servlet/filterlistener/VowelRemoverFilter.java]: It was a mantra in English class: "The vowels are A, E, I, O, U and sometimes Y"

I considered randomly removing Y but didn't want to complicate the unimportant details of the example.:)

**Comment by jfuerth** [comment]
Thanks for the review, Pete. I will fix the problems and clarify the stuff that made you scratch your head.

File: servlet-filterlistener/src/main/java/org/jboss/as/quickstarts/servlet/filterlistener/FilterExampleServlet.java (hunk 1 above)
Comment (jfuerth) [inline servlet-filterlistener/src/main/java/org/jboss/as/quickstarts/servlet/filterlistener/FilterExampleServlet.java]: Yes. Thanks for catching that.

File: servlet-filterlistener/src/main/java/org/jboss/as/quickstarts/servlet/filterlistener/VowelRemoverFilter.java (hunk 3 above)
Comment (pmuir) [inline servlet-filterlistener/src/main/java/org/jboss/as/quickstarts/servlet/filterlistener/VowelRemoverFilter.java]: Hehe.

**Comment by csadilek** [comment]
Pete, the comments have been fixed now and all commits have been collapsed into the two listed above.

**Comment by pmuir** [comment]
Merged. THanks guys

**Pull request closed without merging**